org_fedora_oscap/rule_handling.py

```diff
@@ -752,3 +782,258 @@ def revert_changes(self, ksdata, storage):
+        """:see: RuleHandler.eval_rules"""
+
+        messages = []
+
+        if self._firewall_default_enabled is None:
+            # firewall default startup setting
+            self._firewall_default_enabled = ksdata.firewall.enabled
+
+        if self._firewall_enabled is False:
+            msg = _("Firewall will be disabled on startup")
+            messages.append(RuleMessage(self.__class__,
+                                        common.MESSAGE_TYPE_INFO, msg))
+            if not report_only:
+                ksdata.firewall.enabled = self._firewall_enabled
+
+        elif self._firewall_enabled is True:
+            msg = _("Firewall will be enabled on startup")
+            messages.append(RuleMessage(self.__class__,
+                                        common.MESSAGE_TYPE_INFO, msg))
+            if not report_only:
+                ksdata.firewall.enabled = self._firewall_enabled
+
+        # add messages for the already added services
+        for svc in self._added_svcs:
+            msg = _("service '%s' has been added to the list of services to be "
+                    "added to the firewall" % svc)
+            messages.append(RuleMessage(self.__class__,
+                                        common.MESSAGE_TYPE_INFO, msg))
+
+        # add messages for the already added ports
+        for port in self._added_ports:
+            msg = _("port '%s' has been added to the list of ports to be "
+                    "added to the firewall" % port)
+            messages.append(RuleMessage(self.__class__,
+                                        common.MESSAGE_TYPE_INFO, msg))
+
+        # add messages for the already added trusts
+        for trust in self._added_trusts:
+            msg = _("trust '%s' has been added to the list of trusts to be "
+                    "added to the firewall" % trust)
+            messages.append(RuleMessage(self.__class__,
+                                        common.MESSAGE_TYPE_INFO, msg))
+
+        # services, that should be added
+        services_to_add = (svc for svc in self._add_svcs
+                           if svc not in ksdata.firewall.services)
+
+        # ports, that should be added
+        ports_to_add = (ports for ports in self._add_ports
+                        if ports not in ksdata.firewall.ports)
+
+        # trusts, that should be added
+        trusts_to_add = (trust for trust in self._add_trusts
+                         if trust not in ksdata.firewall.trusts)
+
+        for svc in services_to_add:
+            # add the service unless already added
+            if not report_only:
+                self._added_svcs.add(svc)
+                ksdata.firewall.services.append(svc)
+
+            msg = _("service '%s' has been added to the list of services to be "
+                    "added to the firewall" % svc)
+            messages.append(RuleMessage(self.__class__,
+                                        common.MESSAGE_TYPE_INFO, msg))
```

Comment:
I took another look at the code here and close to the line No. 917. I write remarks from both places here:
 - In the code here, a message is printed for every service in `services_to_add`, even if they are already present in `self._added_svcs` and not really "added".
- However, it also seems to me that `self._added_svcs` is empty, as it becomes populated in the code here. Or is there a chance that it would hold some services? If not, the code around line 917 could go away.
- In case when a service is present both in `self._added_svcs` and `services_to_add`, an identical message would be generated twice for every service.

Reply:
@matejak after some testing, line 917 is for anaconda gui messages when a profile is selected with anaconda on install and line 955 for unit test messages (`make test`). From what I can tell, line 955 is only run in unit testing and not when selecting a profile in the OAA GUI. I tested this by adding a different number for each of the messages.

BTW. Thanks for reviewing and looking at this @matejak!